internal/controller/postgrescluster/instance.go

```diff
@@ -1200,25 +1200,34 @@ func (r *Reconciler) reconcileInstance(
 			spec, instanceCertificates, instanceConfigMap, &instance.Spec.Template)
 	}
 
+	// If either OTel feature gates are enabled
 	if err == nil &&
-		(feature.Enabled(ctx, feature.OpenTelemetryLogs) && !feature.Enabled(ctx, feature.OpenTelemetryMetrics)) {
+		(feature.Enabled(ctx, feature.OpenTelemetryLogs) ||
+			feature.Enabled(ctx, feature.OpenTelemetryMetrics)) {
+		{
 
-		// TODO: Setting the includeLogrotate argument to false for now. This
-		// should be changed when we implement log rotation for postgres
-		collector.AddToPod(ctx, cluster.Spec.Instrumentation, cluster.Spec.ImagePullPolicy, instanceConfigMap, &instance.Spec.Template.Spec,
-			[]corev1.VolumeMount{postgres.DataVolumeMount()}, "", false)
-	}
+			// If OTel metrics are enabled, we'll need the MonitoringUserSecret
+			var password = ""
+			if feature.Enabled(ctx, feature.OpenTelemetryMetrics) {
+				monitoringUserSecret := &corev1.Secret{ObjectMeta: naming.MonitoringUserSecret(cluster)}
+				err = errors.WithStack(
+					r.Client.Get(ctx, client.ObjectKeyFromObject(monitoringUserSecret), monitoringUserSecret))
 
-	if err == nil &&
-		feature.Enabled(ctx, feature.OpenTelemetryMetrics) {
+				if err == nil {
+					password = string(monitoringUserSecret.Data["password"])
+				}
+			}
 
-		monitoringUserSecret := &corev1.Secret{ObjectMeta: naming.MonitoringUserSecret(cluster)}
-		err = errors.WithStack(
-			r.Client.Get(ctx, client.ObjectKeyFromObject(monitoringUserSecret), monitoringUserSecret))
+			var logrotate = false
+			if pgbackrest.RepoHostVolumeDefined(cluster) {
```

Comment:
```suggestion
			if feature.Enabled(ctx, feature.OpenTelemetryLogs) && pgbackrest.RepoHostVolumeDefined(cluster) {
```

Reply:
Gonna probably have to do some rebasing here at some point